lib/pure/asyncfutures.nim

```diff
@@ -30,7 +34,14 @@ type
 when not defined(release):
   var currentID = 0
 
-proc callSoon*(cbproc: proc ()) {.gcsafe.}
+var callSoonProc* {.threadvar.}: (proc(cbproc: proc ()) {.gcsafe.})
```

Comment:
I think it would make sense to follow the same convention as the dispatcher's:

* A ``setCallSoonProc`` procedure, and
* a ``getCallSoonProc`` procedure